tools/moFF/moff.xml

```diff
@@ -0,0 +1,351 @@
+                </when>
+            </conditional>
+        </xml>
+        <!-- tokens (code snippets used in <command>) -->
+        <token name="@IDENT_INPUT_ARG_MULTIPLE@"><![CDATA[
+            ## this is where the ident input gets passed to moff/moff_all/moff_mbr
+            #if ($task.ident_input.input_type_selector == "ps"):
+                --inputtsv
+                #for $counter,$filename in enumerate($task.ident_input.psm_input):
+                    './ident_inputs/${filename.display_name}'
+            #end for
+            #else
+                --inputtsv
+                #for $counter,$filename in enumerate($task.ident_input.generic_input):
+                    './ident_inputs/${filename.display_name}'
+                #end for
+            #end if
+        ]]></token>
+        <token name="@IDENT_INPUT_ARG_SINGLE@"><![CDATA[
+            ## this is where the ident input gets passed to moff/moff_all/moff_mbr
+            #if ($task.ident_input.input_type_selector == "ps"):
+                --inputtsv './ident_inputs/${task.ident_input.psm_input.display_name}'
+            #else
+                --inputtsv './ident_inputs/${task.ident_input.generic_input.display_name}'
+            #end if
+        ]]></token>
+        <token name="@WRANGLE_IDENT_INPUT_SINGLE@"><![CDATA[
+            mkdir ./ident_inputs &&
+            #if ($task.ident_input.input_type_selector == "ps"):
+                ln -s '$task.ident_input.psm_input' './ident_inputs/$task.ident_input.psm_input.display_name' &&
+            #else
+                ## optionally remove first line
+                #if ($task.ident_input.remove_header):
+                    sed -i '1d' '$task.ident_input.generic_input' &&
+                #end if
+                ## header row with correct names: "peptide", "prot", "mod_peptide", "rt", "mz", "mass", and "charge"
+                echo -e "peptide\tprot\tmod_peptide\trt\tmz\tmass\tcharge" > tempfile.tab &&
+                awk 'BEGIN{OFS="\t"; FS="\t"}{print \$pep,\$prot,\$mod,\$rt,\$mz,\$mass,\$charge}' pep="${task.ident_input.peptide}" prot="$task.ident_input.prot" mod="$task.ident_input.mod_peptide" rt="$task.ident_input.rt" mz="$task.ident_input.mz" mass="$task.ident_input.mass" charge="$task.ident_input.charge" '$task.ident_input.generic_input' >> tempfile.tab &&
+                mv tempfile.tab '$task.ident_input.generic_input' &&
+                ln -s '$task.ident_input.generic_input' './ident_inputs/$task.ident_input.generic_input.display_name' &&
+            #end if
+        ]]></token>
+        <token name="@WRANGLE_IDENT_INPUT_MULTIPLE@"><![CDATA[
+            mkdir ./ident_inputs &&
+            #if ($task.ident_input.input_type_selector == "ps"):
+                #for $counter,$filename in enumerate($task.ident_input.psm_input):
+                    ln -s '$filename' './ident_inputs/$filename.display_name' &&
```

Comment:
You don't use the counter here and could potentially overwrite the link, or fail because of a link already exists.

Reply:
What do you suggest to do about the link? That would happen if 2 or more datasets had the same display name, right? That would pretty much be an unresolvable problem because the only way that moFF associates an ID file with spectrum file is by name.

Reply:
You could do something like this:
```
#for $counter,$filename in enumerate($task.ident_input.psm_input):
     ln -s '$filename' './ident_inputs/$counter_$filename.display_name' &&
#end for
```

Reply:
>  file is by name

You mean by filename on the hard-drive? If so you could use the counter, no?

Reply:
I don't think we could use the counter, because if the ID files and spectrum files have the same names but are provided in a different order there will be unmatched files.

moFF requires files to have the same name on the hard drive, which is the reason that we have to rename them by the display name.